test/billGenerationTests.test.js

```diff
@@ -0,0 +1,164 @@
+      {statementNumber: 1},
+      {statementNumber: 2}
+    ];
+    const billsForSubscriberNine = [
+      {statementNumber: 2, startDate: sixtyDaysAgo, endDate: thirtyDaysAgo},
+      {statementNumber: 1}
+    ];
+    const generationDataForSubThree = [5263, 4112, 6023, 4801];
+    const generationDataForSubFive = [9873, 9827, 3648, 8642];
+    const generationDataForSubNine = [1253, 1911, 2144, 1306];
+    const rateScheduleForSubThree = {id: 2, rebateRate: 0.093, rate: 0.18};
+    const rateScheduleForSubFive = {id: 3, rebateRate: 0.048, rate: 0.21};
+    const rateScheduleForSubNine = {id: 4, rebateRate: 1.002, rate: 0.06};
+
+
+    before(async () => {
+      sandbox.stub(request, 'getSolarProjectById').returns(solarProject);
+      const stubbedGetOwnerById = sandbox.stub(request, 'getOwnerById');
+      stubbedGetOwnerById.withArgs(3).returns(subscriberThree);
+      stubbedGetOwnerById.withArgs(5).returns(subscriberFive);
+      stubbedGetOwnerById.withArgs(9).returns(subscriberNine);
+      const stubbedGetPGEBill = sandbox.stub(utilityAPI, 'getLatestPGEBill');
+      stubbedGetPGEBill.withArgs(2).returns(latestPGEBillMeterTwo);
+      stubbedGetPGEBill.withArgs(8).returns(latestPGEBillMeterEight);
+      stubbedGetPGEBill.withArgs(7).returns(latestPGEBillMeterSeven);
+      const stubbedGetSubBullsById = sandbox.stub(request, 'getSubscriberBillsByIds')
+      stubbedGetSubBullsById.withArgs([5,2])
+        .returns(billsForSubscriberThree);
+      stubbedGetSubBullsById.withArgs([6, 1, 4])
+        .returns(billsForSubscriberNine);
+      sandbox.stub(request, 'updateSubscriberBill');
+      const stubbedGetEnphaseData = sandbox.stub(enphase, 'getEnphaseDataForSubscriber');
+      stubbedGetEnphaseData.onCall(0).returns(generationDataForSubThree);
+      stubbedGetEnphaseData.onCall(1).returns(generationDataForSubFive);
+      stubbedGetEnphaseData.onCall(2).returns(generationDataForSubNine);
+      const stubbedGetRateSchedule = sandbox.stub(request, 'getRateScheduleById');
+      stubbedGetRateSchedule.withArgs(2).returns(rateScheduleForSubThree);
+      stubbedGetRateSchedule.withArgs(3).returns(rateScheduleForSubFive);
+      stubbedGetRateSchedule.withArgs(4).returns(rateScheduleForSubNine);
+      sandbox.stub(request, 'createSubscriberBill');
+      sandbox.stub(pdfgeneration, 'generatePdfForSubscriber');
+      sandbox.stub(email, 'sendEmail'); // TODO maybe should throw an error for test
+      await generateBillsForSolarProject(solarProjectId);
+    });
+
+    after(() => {
+      sandbox.restore();
+    });
+
+    it("should get a solar project by its id", async () => {
+      expect(request.getSolarProjectById.called).to.equal(true);
+    });
+
+    it("should get each subscriber associated with the solar project", async () => {
+      expect(request.getOwnerById.called).to.equal(true);
+      expect(request.getOwnerById.calledWith(3)).to.equal(true);
```

Comment:
small detail, but changing this and future statements to say something like:

```      expect(request.getOwnerById.calledWith(subscriberThree.id)).to.equal(true);```

where you use the name of the variable instead of the integer helps with readability, it's not immediately apparent what the integers are meant to refer to